### DISCUSSION TITLE
Is there any way to generate models with pydantic.PrivateAttr fields?

### DISCUSSION THREAD

**Discussion by balu211221** [koxudaxi/datamodel-code-generator#1055] [Q&A] [answered]
Is there any option to generate a field as pydantic.PrivateAttr

Ref: https://docs.pydantic.dev/usage/models/#private-model-attributes

**Comment by koxudaxi** [comment, marked as answer]
@balu211221 
We can't define fields with PrivateAttr.
But, we can generate field name with `_` prefix.
Pydantic supports `Config.underscore_attrs_are_private` for `_` prefix field.
Can this way cover your use-case?

**Reply by koxudaxi** [reply]
You can give the config specific model with `--extra-template-data` option.
https://github.com/koxudaxi/datamodel-code-generator/pull/861/files#diff-96cd3db7657ae012e5e90c9b016d36593ad1dec7cdccc98cb7b963d3be295f25R4

Or, you can uses `--custom-template-dir` option for custom jinja2 template.
https://github.com/koxudaxi/datamodel-code-generator/issues/553

**Reply by balu211221** [reply]
Yes, it will work. Thank you.
